### ISSUE TITLE
Sass vars with importers and multiple files with same name

### ISSUE THREAD

**Issue by kenborge** [sass/node-sass#1231]
I'm using the node-sass-json-importer to parse JSON files into SASS variables and are experiencing some weird behaviour when it comes to variables not overriding when files have the same name in different folders. 

Given the following files:
folder/main.scss
``` sass
@import "../default/importer";
@import "shared-variables.json";
body {
    width: $var;
}
```
folder/shared-variables.json
```json
{
    "var": "2em"
}
```
default/importer.scss
```sass
@import "shared-variables.json";
```
default/shared-variables.json
```json
{
    "var": "1em"
}
```

I would expect the output to be
```css
body {
    width: 2em;
}
```
but instead i am getting
```css
body {
    width: 1em;
}
```

The source map seems to indicate the correct files are loaded
```json
{
	"version": 3,
	"file": "main.css",
	"sources": [
		"../../../assets/styles/folder/main.scss",
		"../../../assets/styles/default/importer.scss",
		"../../../assets/styles/default/shared-variables.json",
		"../../../assets/styles/folder/shared-variables.json"
	],
	"mappings": ";AAKA,IAAI,CAAC;EACD,KAAK,EENH,GAAG,GFOR",
	"names": []
}
```
Now if the folder/shared-variables.json is renamed to say shared-variables2.json everything works fine. I'm pretty confident it's not an issue with the importer as i tried recreating the situation with just a custom importer function with some if else statements that returned variable declarations, and got the same issue as long as the imports have the same name in the different folders

**Comment by chriseppstein** [comment]
This may be related to https://github.com/sass/libsass/issues/1638

**Comment by nschonni** [comment]
This appears to be fixed upstream after libsass 3.3.2 (node-sass 3.4.2+)
